micro_sam/training/util.py

```diff
@@ -210,6 +224,29 @@ def __call__(self, x, y, n_pos, n_neg, get_boxes=False, n_samples=None):
         return batched_inputs, batched_sampled_cell_ids_list
 
 
+class ConvertToSemanticSamInputs:
+    """
+    """
+    def __call__(self, x, y):
+        """Convert the outputs of dataloader to the batched format of inputs expected by SAM.
+        """
+        batched_inputs, downsized_gt = [], []
+        for image, gt in zip(x, y):
+            batched_input = {"image": image, "original_size": image.shape[1:]}
+            batched_inputs.append(batched_input)
+
+            # downsize the labels
+            gt_shape = (gt.shape[0], 256, 256)
```

Comment:
Do we still need `downsized_gt`? I think we don't anymore given the updates to the semantic training.
Can we remove it?

Reply:
Ah yes, I think this should now work without the need of this. Hence, we can have the combined objective of the CrossEntropy (or BCE) and Dice over the `batch_masks` vs the original ground truth shape. I'll take care of this right away. Thanks for spotting.